server/server_test.go

```diff
@@ -184,4 +254,62 @@ func TestHTTPServer_ReportEndpoint(t *testing.T) {
 	}, testTimeout)
 }
 
-// TODO: test more cases for report endpoint
+// TestDisabledInternalOrganizationsBadOrgID tests accessing internal rule when EnableInternalRulesOrganizations = true and request id is not allowed
+func TestDisabledInternalOrganizationsBadOrgID(t *testing.T) {
+	helpers.RunTestWithTimeout(t, func(t *testing.T) {
+
+		loadMockRuleContentDir(RuleContentInternal1)
+
+		_, err := content.GetRuleContent(internalTestRuleModule)
+		helpers.FailOnError(t, err)
+
+		helpers.AssertAPIRequest(t, &serverConfigInternalOrganizations, &helpers.DefaultServicesConfig, nil, &helpers.APIRequest{
+			Method:             http.MethodGet,
+			Endpoint:           server.RuleContent,
+			EndpointArgs:       []interface{}{internalTestRuleModule},
+			AuthorizationToken: "Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhY2NvdW50X251bWJlciI6IjUyMTM0NzYiLCJvcmdfaWQiOiIxMjM0In0.Y9nNaZXbMEO6nz2EHNaCvHxPM0IaeT7GGR-T8u8h_nr_2b5dYsCQiZGzzkBupRJruHy9K6acgJ08JN2Q28eOAEVk_ZD2EqO43rSOS6oe8uZmVo-nCecdqovHa9PqW8RcZMMxVfGXednw82kKI8j1aT_nbJ1j9JZt3hnHM4wtqydelMij7zKyZLHTWFeZbDDCuEIkeWA6AdIBCMdywdFTSTsccVcxT2rgv4mKpxY1Fn6Vu_Xo27noZW88QhPTHbzM38l9lknGrvJVggrzMTABqWEXNVHbph0lXjPWsP7pe6v5DalYEBN2r3a16A6s3jPfI86cRC6_oeXotlW6je0iKQ",
+		}, &helpers.APIResponse{
+			StatusCode: http.StatusForbidden,
+		})
+	}, testTimeout)
+}
+
+// TestDisabledInternalOrganizationsAllowedOrgID tests accessing internal rule when EnableInternalRulesOrganizations = true and org id is allowed
+func TestDisabledInternalOrganizationsAllowedOrgID(t *testing.T) {
+	helpers.RunTestWithTimeout(t, func(t *testing.T) {
+
+		loadMockRuleContentDir(RuleContentInternal1)
+
+		_, err := content.GetRuleContent(internalTestRuleModule)
+		helpers.FailOnError(t, err)
+
+		helpers.AssertAPIRequest(t, &serverConfigInternalOrganizations, &helpers.DefaultServicesConfig, nil, &helpers.APIRequest{
+			Method:             http.MethodGet,
+			Endpoint:           server.RuleContent,
+			EndpointArgs:       []interface{}{internalTestRuleModule},
+			AuthorizationToken: "Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhY2NvdW50X251bWJlciI6IjUyMTM0NzYiLCJvcmdfaWQiOiIxIiwianRpIjoiMDU0NDNiOTktZDgyNC00ODBiLWE0YmUtMzc5Nzc0MDVmMDkzIiwiaWF0IjoxNTk0MTI2MzQwLCJleHAiOjE1OTQxNDE4NDd9.pp32mPoypnRjOYE95SrBar0fdLS9t_hndOtP5qUvB-c",
+		}, &helpers.APIResponse{
+			StatusCode: http.StatusOK,
+		})
+	}, testTimeout)
+}
+
+// TestDisabledInternalOrganizations tests accessing internal rule when EnableInternalRulesOrganizations = false
+func TestDisabledInternalOrganizations(t *testing.T) {
+	helpers.RunTestWithTimeout(t, func(t *testing.T) {
+
+		loadMockRuleContentDir(RuleContentInternal1)
+
+		_, err := content.GetRuleContent(internalTestRuleModule)
+		helpers.FailOnError(t, err)
+
+		helpers.AssertAPIRequest(t, &serverConfigJWT, &helpers.DefaultServicesConfig, nil, &helpers.APIRequest{
```

Comment:
Btw, these tests could be refactored by using `t.Run("sub test name", func...`

Reply:
oh right, thanks! done